packages/sdk/src/modules/rbac/index.ts

```diff
@@ -0,0 +1,95 @@
+import { v4 as uuidv4, v4 } from 'uuid';
+import { CreateStorageKeysEnum, FetchRoles } from '../../types';
+import { Base } from '../base';
+import { ApiPromise } from '@polkadot/api';
+import { createStorageKeys } from '../../utils';
+import { stringToU8a } from '@polkadot/util';
+import type { SDKMetadata, Address } from '../../types';
+import type { CodecHash } from '@polkadot/types/interfaces/runtime/types';
+import { Entity } from '@peaq-network/types/interfaces';
+
+interface CreateNewRole {
+  name: string;
+  id?: string;
+  address?: Address;
+  seed?: string;
+}
+
+export class RBAC extends Base {
+  constructor(
+    protected override readonly _api?: ApiPromise,
+    protected readonly _metadata?: SDKMetadata
+  ) {
+    super();
+  }
+  /**
+   * Creates a new role.
+   * @param name - The options for creating the Roles.
+   * @returns A promise that resolves when the role is created.
+   */
+
+  public async createRole(options: CreateNewRole) {
+    try {
+      const { name, id = '', address = '', seed = '' } = options;
+      if (!name) throw new Error('Name is required');
+      const generatedId = v4();
+      const roleId = stringToU8a(id || generatedId);
+      const roleName = stringToU8a(name);
+      const api = this._getApi();
+      const keyPair = this._metadata?.pair || this._getKeyPair(seed);
+      const addRoleExtrinsics = api.tx?.['peaqRbac']?.['addRole'](
+        roleId,
+        roleName
+      );
+      const nonce = await this._getNonce(address || keyPair.address);
+      await this._signTransaction({
+        nonce,
+        address: keyPair,
+        extrinsics: addRoleExtrinsics,
+      });
+      return {
+        hash: addRoleExtrinsics.hash as unknown as CodecHash,
+        id: id || generatedId,
+      };
+    } catch (error) {
+      throw new Error(`Error creating role: ${error}`);
+    }
+  }
+
+  /**
+   * Fetch all roles.
+   * @param name - The ownerAddress is public address of user or owner who created a roles.
```

Comment:
wrong param name: "name" should be changed to "ownerAddress"

Please check all your codes for this and correct them.

Reply:
done